src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/SphKerrSchild.cpp

```diff
@@ -0,0 +1,1053 @@
+    }
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3, Frame>*> G1_dot_x,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::G1_dot_x<DataType, Frame> /*meta*/) const {
+  const auto& x_minus_center =
+      cache->get_var(*this, internal_tags::x_minus_center<DataType, Frame>{});
+  const auto& helper_matrix_G1 =
+      cache->get_var(*this, internal_tags::helper_matrix_G1<DataType, Frame>{});
+
+  for (size_t i = 0; i < 3; ++i) {
+    G1_dot_x->get(i) = helper_matrix_G1.get(i, 0) * x_minus_center.get(0);
+    for (size_t m = 1; m < 3; ++m) {
+      G1_dot_x->get(i) += helper_matrix_G1.get(i, m) * x_minus_center.get(m);
+    }
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::i<DataType, 3, Frame>*> G2_dot_x,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::G2_dot_x<DataType, Frame> /*meta*/) const {
+  const auto& x_minus_center =
+      cache->get_var(*this, internal_tags::x_minus_center<DataType, Frame>{});
+  const auto& helper_matrix_G2 =
+      cache->get_var(*this, internal_tags::helper_matrix_G2<DataType, Frame>{});
+
+  for (size_t j = 0; j < 3; ++j) {
+    G2_dot_x->get(j) = helper_matrix_G2.get(0, j) * x_minus_center.get(0);
+    for (size_t n = 1; n < 3; ++n) {
+      G2_dot_x->get(j) += helper_matrix_G2.get(n, j) * x_minus_center.get(n);
+    }
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::Ij<DataType, 3, Frame>*> inv_jacobian,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::inv_jacobian<DataType, Frame> /*meta*/) const {
+  const auto& transformation_matrix_Q = cache->get_var(
+      *this, internal_tags::transformation_matrix_Q<DataType, Frame>{});
+  const auto& G1_dot_x =
+      cache->get_var(*this, internal_tags::G1_dot_x<DataType, Frame>{});
+  const auto& G2_dot_x =
+      cache->get_var(*this, internal_tags::G2_dot_x<DataType, Frame>{});
+
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      inv_jacobian->get(j, i) =
+          transformation_matrix_Q.get(i, j) + G1_dot_x.get(i) * G2_dot_x.get(j);
+    }
+  }
+}
+
+template <typename DataType, typename Frame>
+void SphKerrSchild::IntermediateComputer<DataType, Frame>::operator()(
+    const gsl::not_null<tnsr::Ij<DataType, 3, Frame>*> helper_matrix_E1,
+    const gsl::not_null<CachedBuffer*> cache,
+    internal_tags::helper_matrix_E1<DataType, Frame> /*meta*/) const {
+  const auto& rho = get(cache->get_var(*this, internal_tags::rho<DataType>{}));
+  const auto& r_squared =
+      get(cache->get_var(*this, internal_tags::r_squared<DataType>{}));
+  const auto spin_a = solution_.dimensionless_spin() * solution_.mass();
```

Comment:
consider factoring this and the next computation into a free function
```cpp
const auto [spin_a, a_squared] = spin_a_and_squared(solution_);
```
Function can return `std::tuple<spin_a_type, double>`

The `[a,a_squ]` thing is called "structured bindings"

Reply:
Also added function for spin_a by itself